### ISSUE TITLE
Add option to disable "already validated" feature

### ISSUE THREAD

**Issue by nh2** [lukas2511/dehydrated#370]
Release 0.4.0 adds the `Skip challenges for already validated domains` feature.

While the feature makes sense, this makes it difficult to test hooks.

Please add an option to disable this feature (or make skipping it part of `--force`).

**Comment by txr13** [comment]
@nh2 Does this happen in staging as well?

**Comment by nh2** [comment]
> Does this happen in staging as well?

Yes it says:

```
 + Requesting challenge for mydomain.com...
 + Already validated!
```

**Comment by lukas2511** [comment]
Mh, sorry, but if you need to test a hook just call it manually with the corresponding parameters or use different subdomains to force validation.

**Comment by widder** [comment]
How to work around this? Need to force this tool really quick as my certificates expire

**Comment by widder** [comment]
Here is solution: just add this to daily cron:

rm -rf /etc/dehydrated/accounts/*
rm -rf /etc/dehydrated/certs/*

**Comment by txr13** [comment]
@widder What is the issue you're experiencing? The "already validated" response will not block the renewal of certificates.

And I think it a very bad "solution" to register a brand-new account with the CA every time you want a new certificate...

**Comment by pinkgothic** [comment]
@txr13 I'm not widder, but I might be able to provide some insight into a comparable situation. The problem I've got is trying to get a validation hook to work properly.

The "Already validated!" feature (as excellent and useful as it is in normal situations!) is interfering with repeated testing of the hook against LE's staging network. I can test the hook on its own, but one of the problems it's wrought with is actually its integration with dehydrated - occasionally it will complain it's not being called with the right arguments.

So, basically, I'd like a way to skip validation-caching, too, to try and force more deterministic behaviour out of this hook I'm trying to run. That said, I'm going to see if I can take care of this with a sloppy hack on my end (or less sloppy if I get lucky), so there's hopefully no urgency - but I wanted to explain a possible usecase in this ticket.

tl;dr: Flag would be useful for hook<->dehydrated integration testing against LE's staging network.

**Comment by widder** [comment]
What @pinkgothic says is correct.
I'm not using dehydrated any more because of its inability to use it with staging.

**Comment by txr13** [comment]
@lukas2511 There does seem to be a solid use case for wanting to skip the validation caching. Rather than adding a new flag, what about the suggestion of making --force also request a new validation?

**Comment by lukas2511** [comment]
See referenced commit, this should allow you to do what you are trying to do. Thanks @txr13 for the suggestion.
